src/gui/qgscolorrampbutton.h

```diff
@@ -0,0 +1,323 @@
+    Q_PROPERTY( bool acceptLiveUpdates READ acceptLiveUpdates WRITE setAcceptLiveUpdates )
+    Q_PROPERTY( bool showMenu READ showMenu WRITE setShowMenu )
+    Q_PROPERTY( Behaviour behaviour READ behaviour WRITE setBehaviour )
+    Q_PROPERTY( QgsColorRamp* defaultColorRamp READ defaultColorRamp WRITE setDefaultColorRamp )
+    Q_PROPERTY( QString context READ context WRITE setContext )
+
+  public:
+
+    /** Specifies the behaviour when the button is clicked
+     */
+    enum Behaviour
+    {
+      ShowDialog = 0, //!< Show a color ramp dialog when clicked
+      SignalOnly //!< Emit colorRampClicked signal only, no dialog
+    };
+
+    /** Construct a new color ramp button.
+     * @param parent The parent QWidget for the dialog
+     * @param cdt The title to show in the color chooser dialog
+     */
+    QgsColorRampButton( QWidget *parent = nullptr, const QString& cdt = "" );
+
+    virtual ~QgsColorRampButton();
+
+    virtual QSize sizeHint() const override;
+
+    /** Return the current color ramp.
+     * @returns currently selected color
+     * @see setColor
+     */
+    QgsColorRamp* colorRamp() const;
+
+    /** Set the title for the color ramp dialog window.
+     * @param title Title for the color ramp dialog
+     * @see colorRampDialogTitle
+     */
+    void setColorRampDialogTitle( const QString& title );
+
+    /** Returns the title for the color ramp dialog window.
+     * @returns title for the color ramp dialog
+     * @see setColorRampDialogTitle
+     */
+    QString colorRampDialogTitle() const;
+
+    /** Returns whether the button accepts live updates from QColorRampDialog.
+     * @returns true if the button will be accepted immediately when the dialog's color changes
+     * @see setAcceptLiveUpdates
+     */
+    bool acceptLiveUpdates() const { return mAcceptLiveUpdates; }
+
+    /** Sets whether the button accepts live updates from QColorRampDialog. Live updates may cause changes
+     * that are not undoable on QColorRampDialog cancel.
+     * @param accept set to true to enable live updates
+     * @see acceptLiveUpdates
+     */
+    void setAcceptLiveUpdates( const bool accept ) { mAcceptLiveUpdates = accept; }
+
+    /** Sets whether the drop down menu should be shown for the button. The default behaviour is to
+     * show the menu.
+     * @param showMenu set to false to hide the drop down menu
+     * @see showMenu
+     */
+    void setShowMenu( const bool showMenu );
+
+    /** Returns whether the drop down menu is shown for the button.
+     * @returns true if drop down menu is shown
+     * @see setShowMenu
+     */
+    bool showMenu() const { return menu() ? true : false; }
+
+    /** Sets the behaviour for when the button is clicked. The default behaviour is to show
+     * a color ramp dialog.
+     * @param behaviour behaviour when button is clicked
+     * @see behaviour
+     */
+    void setBehaviour( const Behaviour behaviour );
```

Comment:
Ah - then it's definitely not needed here. That was just a (hack?) to allow the color button to be reused inside the color picker dialog and avoid multiple overlapping color pickers.